src/core/handlers/RequestHandler.ts

```diff
@@ -200,7 +204,16 @@ export abstract class RequestHandler<
 
     // Clone the request instance before it's passed to the handler phases
     // and the response resolver so we can always read it for logging.
-    const mainRequestRef = args.request.clone()
+    if (!RequestHandler.#mainRequestRefCache.has(args.request)) {
```

Comment:
This is great but a bit verbose. What do you think if we abstract it to a protected method on the `RequestHandler` class itself? Something like `.cloneRequest()`?